lms/djangoapps/grades/new/course_grade.py

```diff
@@ -145,8 +203,22 @@ def compute_and_update(self, read_only=False):
         subsections_created = len(subsection_grade_factory._unsaved_subsection_grades)  # pylint: disable=protected-access
         subsections_read = subsections_total - subsections_created
         blocks_total = len(self.locations_to_scores)
-        if not read_only:
+        if not read_only and self.has_access_to_course:
```

Comment:
Why does access to course have to be checked here?  We shouldn't have even gotten this far if access to the course is not allowed.